deepfence_worker/tasks/sbom/generate_sbom.go

```diff
@@ -124,6 +116,15 @@ func (s SbomGenerator) GenerateSbom(ctx context.Context, task *asynq.Task) error
 		return err
 	}
 
+	log.Info().Msgf("Adding scan id to map:%s", params.ScanId)
+	scanMap.Store(params.ScanId, scanCtx)
+	defer func() {
+		log.Info().Msgf("Removing scan id from map:%s", params.ScanId)
+		scanMap.Delete(params.ScanId)
+		res <- err
+		close(res)
+	}()
+
```

Comment:
This needs to be right after we acquire `res` line 87 or we will be leaking go routines in case we return before sending to `res`

Reply:
in that place the scan id is not known to to as its not yest decoded

Reply:
the order was wrong updated in latest commit